UnityProject/Assets/Scripts/Objects/Construction/WindowObject.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using Mirror;
+using Random = UnityEngine.Random;
+
+/// <summary>
+/// The main girder component
+/// </summary>
+public class WindowObject : NetworkBehaviour, ICheckedInteractable<HandApply>
+{
+	private TileChangeManager tileChangeManager;
+
+	private RegisterObject registerObject;
+	private ObjectBehaviour objectBehaviour;
+
+	[Tooltip("Layer tile which this will create when placed.")]
+	public LayerTile layerTile;
+
+	private void Start()
+	{
+		tileChangeManager = GetComponentInParent<TileChangeManager>();
+		registerObject = GetComponent<RegisterObject>();
+		GetComponent<Integrity>().OnWillDestroyServer.AddListener(OnWillDestroyServer);
+		objectBehaviour = GetComponent<ObjectBehaviour>();
+	}
+
+
+	private void OnWillDestroyServer(DestructionInfo arg0)
+	{
+		Spawn.ServerPrefab("GlassShard", gameObject.TileWorldPosition().To3Int(), transform.parent, count: Random.Range(2, 4),
+			scatterRadius: Random.Range(1, 2), cancelIfImpassable: true);
```

Comment:
same Random.Range(int, int) thing

Reply:
What log wrote about screwdriving?
Is it only screwdriving not working?

Reply:
no exceptions about screwdriving;
other tools like wrench didn't work on windows either

Reply:
Did you try to break it?
Is it pushable?

Reply:
see comment below, interactions w/pushable object are fine

Reply:
Screwdriver trait is fucked. In the screwdriver prefab change ScrewDriver trait to Screwdriver. Now it works.
It works with wrench, check again. Window must be unscrewed for wrench to work

Reply:
Window.asset is missin reference to screwdriving...